pkg/http/api.go

```diff
@@ -489,10 +489,10 @@ func (a *api) onOutputBindingMessage(reqCtx *fasthttp.RequestCtx) {
 			req.Metadata = map[string]string{}
 		}
 		// if sc is not empty context, set traceparent Header.
-		if sc != (trace.SpanContext{}) {
+		if !sc.Equal(trace.SpanContext{}) {
 			req.Metadata[traceparentHeader] = diag.SpanContextToW3CString(sc)
 		}
-		if sc.Tracestate != nil {
+		if sc.TraceState().Len() == 0 {
```

Comment:
Should we add a nil-check for `sc.TraceState()`? Or is that guaranteed to be non-nil?

Reply:
@ItalyPaleAle sc.TraceState() returns a struct which as far as I know it cannot be nul. Lenght is the way I figure out to check whether tracestate is empty. Hope that makes sense.